pkg/interface/src/views/apps/links/components/LinkItem.tsx

```diff
@@ -0,0 +1,154 @@
+  api: GlobalApi;
+  group: Group;
+  contacts: Rolodex;
+}
+
+export const LinkItem = (props: LinkItemProps) => {
+  const {
+    node,
+    resource,
+    hideAvatars,
+    hideNicknames,
+    remoteContentPolicy,
+    api,
+    group,
+    contacts,
+    ...rest
+  } = props;
+
+  const URLparser = new RegExp(
+    /((?:([\w\d\.-]+)\:\/\/?){1}(?:(www)\.?){0,1}(((?:[\w\d-]+\.)*)([\w\d-]+\.[\w\d]+))){1}(?:\:(\d+)){0,1}((\/(?:(?:[^\/\s\?]+\/)*))(?:([^\?\/\s#]+?(?:.[^\?\s]+){0,1}){0,1}(?:\?([^\s#]+)){0,1})){0,1}(?:#([^#\s]+)){0,1}/
+  );
+
+  const author = node.post.author;
+  const index = node.post.index.split('/')[1];
+  const size = node.children ? node.children.size : 0;
+  const contents = node.post.contents;
+  const hostname = URLparser.exec(contents[1].url) ? URLparser.exec(contents[1].url)[4] : null;
+
+  const baseUrl = props.baseUrl || `/~404/${resource}`;
+
+  const ourRole = group ? roleForShip(group, window.ship) : undefined;
+  const [ship, name] = resource.split('/');
+
+  const [locationText, setLocationText] = useState('Copy Link Location');
+
+  const copyLocation = () => {
+    setLocationText('Copied');
+    writeText(contents[1].url);
+    setTimeout(() => {
+      setLocationText('Copy Link Location');
+    }, 2000);
+  };
+
+  const deleteLink = () => {
+    if (confirm('Are you sure you want to delete this link?')) {
+      api.graph.removeNodes(`~${ship}`, name, [node.post.index]);
+    }
+  };
+
+  return (
+    <Box width="100%" {...rest}>
+    
+      <Box
+        lineHeight="tall"
+        display='flex'
+        flexDirection='column'
+        width="100%"
+        color='washedGray'
+        border={1}
+        borderRadius={2}
+        alignItems="flex-start"
+        overflow="hidden"
+      >
+        <RemoteContent
+          url={contents[1].url}
+          text={contents[0].text}
+          remoteContentPolicy={remoteContentPolicy}
+          style={{ marginTop: '-1px' }}
+          oembedProps={{
+            p: 2
+          }}
+          textProps={{
+            overflow: 'hidden',
+            color: 'black',
+            display: 'block',
+            style: { textOverflow: 'ellipsis', whiteSpace: 'pre' },
+            p: 2
+          }} />
+        <Text color="gray" p={2} flexShrink={0}>
+          <Anchor target="_blank" rel="noopener noreferrer" style={{ textDecoration: 'none' }} href={contents[1].url}>
+            <Box display='flex'>
+              <Icon icon='ArrowExternal' mr={1} />{hostname}
+            </Box>
+          </Anchor>
+        </Text>
+      </Box>
+      
+      <Row minWidth='0' flexShrink={0} width="100%" justifyContent="space-between" py={3} bg="white">
+      
+      <Author
+        showImage
+        contacts={contacts}
+        ship={author}
+        date={node.post['time-sent']}
+        hideAvatars={hideAvatars}
+        hideNicknames={hideNicknames}
+        remoteContentPolicy={remoteContentPolicy}
+        group={group}
+        api={api}
```

Comment:
You're going to have to weave `history` in here because ProfileOverlay expects `history` for redirecting to the DM.

So you'll have to pass `history` from LinkResource to LinkItem to Author to OverlaySigil to ProfileOverlay, and so on anywhere you're using Author.

Reply:
_case for state hooks intensifies_